### PR TITLE
Replace Authenticator generated Account name from AppConfig.name to S…

### PR DESCRIPTION
**Pull request by jlhughes**
…MTPEmailService.displayName

Issue: [5286](https://github.com/nanoPayinc/NANOPAY/issues/5286)

The AppConfig.name is not suitable for this use as it is one: development, branch_anme, test, staging, demo, prod
While the SMTPEmailService.displayName is 'nanopay Corporation' (default), and 'Ablii' for ablii groups and when logged in under the ablii url.

NP-5286 #time 30m

### CODE REVIEW

**Review by TW80000 — APPROVED** [review]
Did not test, but the code LGTM.

File: src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js [hunk 1]
```diff
@@ -68,8 +69,10 @@ foam.CLASS({
 
         try {
           AppConfig config = (AppConfig) x.get("appConfig");
-          String path = String.format("/%s:%s", config.getName(), user.getEmail());
-          String query = String.format("secret=%s&issuer=%s&algorithm=%s", key, config.getName(), getAlgorithm());
+          SMTPEmailService service = (SMTPEmailService) x.get("smtpEmailService");
+          String name = service == null ? "nanopay Corporation" : service.getDisplayName();
```
Comment (TW80000) [inline src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js]: Oh wait, "nanopay Corporation" shouldn't be in FOAM.

**Review by TW80000 — CHANGES_REQUESTED** [review]

File: src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js]: yes, good call.

Comment (jlhughes) [reply src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js]: @TW80000 Replaced nanopay specific text with 'FOAM'.  Expect our nanopay side to always be configured in production.

**Review by TW80000 — APPROVED** [review]

**Pull request merged**